v2/tools/generator/internal/astmodel/internal_package_reference.go

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package astmodel
+
+// InternalPackageReference describes a package reference that points to a local package (either a storage package
+// or a standard one). It can be used to abstract across the exact package type (storage vs local)
+type InternalPackageReference interface {
+	PackageReference
+
+	// LocalPathPrefix returns the prefix (everything up to the group name)
+	LocalPathPrefix() string
+	Group() string
```

Comment:
You embed PackageReference but then redefine a bunch of its methods here too? Not following why?

Reply:
This used to be `LocalLikePackageReference` - I'm pulling it into the object model fully. I'll simplify the declaration.